Adds AutoOps and Stack Monitoring comparison (#1711)
elastic/docs-content: deploy-manage/monitor/autoops/autoops-stack-monitoring-comparison.md

> ---

Author: That makes sense! The comparison should be a level above both products' pages

> ---

Reviewer: > strongly suggest that this page is brought up a level instead of nested under AutoOps.

@shainaraskas @wajihaparvez - where do you recommend to add the page?

> ---

Author: Added under Monitoring, after the AutoOps and Stack Monitoring sections

applies_to:
  deployment:
    ess: all
    ece: all
    eck: all
    self: all

> navigation_title: AutoOps and Stack Monitoring comparison

Author: I like it

---

# AutoOps and Stack Monitoring comparison

This page provides a detailed comparison of AutoOps and Stack Monitoring to help you decide which solution is better suited to your needs.

## Review the main differences

> Review how both tools differ in their provisioning, setup, access, and investigative abilities.

Reviewer 2: ```suggestion
Review how these tools differ in their provisioning, set up procedure, method of access, and capabilities.
```
 - "both" reads a little oddly for me, since you couldn't really compare just one tool.
 - I'd remove "investigative", because otherwise "abilities" or "capabilities could be interpreted as applying to each noun in the list, i.e. "provisiioning ability", "setup abillity, etc."
 - "capabilities" strikes me as more fitting for something a machine does, and "abilities" for something a person does, but maybe that's just me. Just a guess. :-)

### Resource provisioning and billing

#### AutoOps [ao-resource]
AutoOps stores and backs up your monitoring data internally on {{ecloud}} infrastructure so you don’t need to think about provisioning, sizing, and availability. The data is retained for four days by default. Using AutoOps is free for {{ecloud}} customers and is offered to [all subscription tiers](https://www.elastic.co/subscriptions/cloud).

#### Stack Monitoring [sm-resource]

> With Stack Monitoring, you are responsible for storing your monitoring data. This means provisioning the necessary resources based on your performance and retention needs as well as paying for the allocated resources. The default retention period is six days.

Reviewer 2: ```suggestion
With Stack Monitoring, you are responsible for storing your monitoring data. This requires provisioning the necessary resources based on your performance and retention needs as well as paying for the allocated resources. The default retention period is six days.
```
For a non-native English speaker "means" might be a tad ambiguous.

### Setup

#### AutoOps [ao-setup]
On {{ech}} (ECH), AutoOps is set up and enabled automatically in all [supported regions](ec-autoops-regions.md) and there is no action required from you.

:::{note}
AutoOps will be available in self-managed environments in the future.
:::

:::{image} /deploy-manage/images/cloud-autoops-setup.png
:alt: Diagram showing AutoOps setup in Elastic Cloud
:::

#### Stack Monitoring [sm-setup]

> Stack Monitoring is a Kibana application that can be enabled on self-managed clusters on your premises, ECH deployments, {{eck}} (ECK), and {{ece}} (ECE). Stack Monitoring is not available on {{serverless-full}} since Elastic takes care of monitoring and managing your Serverless projects.

Reviewer 2: ```suggestion
Stack Monitoring is a {{kib}} application that can be enabled on self-managed clusters on your premises, ECH deployments, {{eck}} (ECK), and {{ece}} (ECE). Stack Monitoring is not available on {{serverless-full}} since Elastic takes care of monitoring and managing your Serverless projects.
```

> Depending on your deployment model, there is [some setup](/deploy-manage/monitor/stack-monitoring.md#configure-and-use-stack-monitoring) involved to enable Stack Monitoring. You need to configure an agent, specify which logs and metrics you want to collect from all your Elastic Stack components, and where to send them.

Reviewer 2: ```suggestion
Depending on your deployment model, there is [some setting up](/deploy-manage/monitor/stack-monitoring.md#configure-and-use-stack-monitoring) involved to enable Stack Monitoring. You need to configure an agent, specify which logs and metrics you want to collect from all your Elastic Stack components, and where to send them.
```

#### Stack Monitoring [sm-availability]
Once setup is complete, you can access the Stack Monitoring UI inside {{kib}}, where you can monitor all your Elastic Stack components.

### Investigations and root cause analysis

#### AutoOps [ao-investigations]
The AutoOps agent captures a pre-defined set of {{es}} metrics, but doesn’t fetch any logs. AutoOps then performs multi-metrics analysis and correlations to identify issues and potential root causes. When issues are detected, AutoOps raises [events](ec-autoops-events.md) and [notifies](ec-autoops-notifications-settings.md) you accordingly. When the issue is resolved, AutoOps automatically closes the event.

For each raised event, AutoOps provides insights into the affected resources (cluster, node, index, shard, etc.), background information on the detected problem, and step-by-step guides to help you diagnose and remediate the identified issues. Most detection rules can be [customized](ec-autoops-event-settings.md) by adjusting thresholds, durations, index patterns, data tiers and more.

#### Stack Monitoring [sm-investigations]

> The Stack Monitoring UI displays the [metrics](/deploy-manage/monitor/monitoring-data/elasticsearch-metrics.md) of your monitored Elastic Stack components over time. Logs can be viewed, searched, and filtered in Discover. You can enable a pre-defined set of alerts that are triggered when specific thresholds are crossed. You can also configure your own [alerts](/deploy-manage/monitor/monitoring-data/configure-stack-monitoring-alerts.md) on any collected metrics or log messages. However, Stack Monitoring does not offer any further investigation, correlations, or root cause analyses.

Reviewer 2: ```suggestion
The Stack Monitoring UI displays the [metrics](/deploy-manage/monitor/monitoring-data/elasticsearch-metrics.md) of your monitored {{stack}} components over time. Logs can be viewed, searched, and filtered in Discover. You can enable a pre-defined set of alerts that are triggered when specific thresholds are crossed. You can also configure your own [alerts](/deploy-manage/monitor/monitoring-data/configure-stack-monitoring-alerts.md) on any collected metrics or log messages. However, Stack Monitoring does not offer any further investigation, correlations, or root cause analyses.
```
